stata_kernel/kernel.py

```diff
@@ -116,13 +126,82 @@ def do_execute(
 
         """
 
-        cm = CodeManager(re.sub(r'\r\n', r'\n', code))
+        #############
+        #  Magics!  #
+        #############
+
+        magic, code = poor_mans_magic(code)
+        quit_early  = {'execution_count': self.execution_count}
+        quit_early['status'] = 'ok'
+        quit_early['payload'] = []
+        quit_early['user_expressions'] = {}
+
+        if magic['name'] in stata_magics['global_magics']:
+            obj      = self.do("macro dir")
+            res      = obj.get('res').rstrip()
+            sglobals = magic['regex'].findall(res)
+            lens     = 0
+            pglobals = []
```

Comment:
I haven't added a `Contributing.md` yet, but can you install [editorconfig](https://editorconfig.org/) and [yapf](https://github.com/google/yapf)? They'll use the project-level configuration files and make all code consistent, and won't make you touch any of your other projects.